R/call_slack_internals.R

```diff
@@ -0,0 +1,150 @@
+#' Lists all channels in a Slack team.
+#'
+#' @inheritParams auth_test
+#' @keywords internal
+#' @noRd
+#' @references https://api.slack.com/methods/conversations.list
+list_channels <- function(bot_user_oauth_token = Sys.getenv("SLACK_BOT_USER_OAUTH_TOKEN"), types = "public_channel", ...) {
+  with_pagination(
+    function() {
+      call_slack_api(
+        "/api/conversations.list",
+        .method = GET,
+        types = types,
+        ...
+      )
+    },
+    extract = "channels"
+  )
+}
+
+
+#' Lists all users in a Slack team.
+#'
+#' @inheritParams auth_test
+#' @keywords internal
+#' @noRd
+#' @references https://api.slack.com/methods/users.list
+list_users <- function(bot_user_oauth_token = Sys.getenv("SLACK_BOT_USER_OAUTH_TOKEN"), ...) {
+  with_pagination(
+    function() {
+      call_slack_api(
+        "/api/users.list",
+        .method = GET,
+        ...
+      )
+    },
+    extract = "members"
+  )
+}
+
+#' Sends a message to a channel.
```

Comment:
is this supposed to supercede `slackr_msg()` / `text_slackr()`?

Reply:
No, not directly.  These functions are all internal.  Thus they have minimal checking for valid inputs and handling exceptions.  The existing exported functions should then make use of these internal functions.